forwards_paper.tex

```diff
@@ -903,59 +903,80 @@ \subsubsection*{Estimates of run-time complexity}
+A particular advantage of this approach is that it enables very
+efficient copying of data, and in principle zero-copy data access
+(where a data consumer reads directly from the memory of a producer) is possible.
 Our implementation uses the NumPy C API~\citep{walt2011numpy} to efficiently copy
 data from Python into the low-level C library used to manipulate tree sequences.
-Since tables are encoded as contiguous arrays of numeric data,
-this achieves data transfer rates that would be impossible under
-any text-based approach while retaining excellent portability.
-
-This also provides basic input and output operations via NumPy's
-array interface, which provides a great deal of flexibility and efficiency,
+This architecture allows for data transfer rates of gigabytes per second
+(impossible under any text-based approach), while retaining excellent portability.
+NumPy's array interface provides a great deal of flexibility and efficiency,
 and makes it straightforward to transfer data from sources
 such as HDF5 \citep{hdf5}or Dask~\citep{dask}.
-For small scale data and debugging purposes, a simple text format is also supported.
-
-% PLR: this was mentioned above, and belongs in the documentation, I think.
-% The API also provides functions to sort tables
-% to ensure that the records are in the form required to input
-% into an \msprime{} tree sequence object. (Simplification may also be required.)
-
-The \msprime{} Tables API is designed to be easily ``plugged'' into any simulation engine.
-One simply needs to fill the relevant columns,
-and to send them to \msprime{} at regular intervals for simplification.
-For example, in the benchmarking simulations above,
-we used the \fwdpp{} \cpp{} API \citep{fwdpp}
-to fill arrays of simple structures representing the node and edge data.
-Doing so requires no modification of the \fwdpp{} code base.
-Rather, we simply need to bookkeep parent/offspring labels,
-and perform simple processing of the recombination breakpoints from each mating event.
-Using \texttt{pybind11} (\url{https://github.com/pybind/pybind11/}),
-the node/edge arrays on the \cpp{} side are made visible, without copy, to Python as a multi-column numpy array,
-which is processed by the \msprime{} tables API.
-Our implementation also uses \texttt{pybind11}
-to create a Python package extending \texttt{fwdpy11}
-(a Python package based on \fwdpp{}; \url{https://github.com/molpopgen/fwdpy11}),
-% (A neat aside, that we may wish to skip here, is that it may be possible to use the tables
-% API from a command-line C/\cpp{} application via an embedded Python interpreter.)
+For small scale data and debugging purposes, a simple text based format is also supported.
+
+The \msprime\ Python Tables API provides a general purpose toolkit for importing
+and processing succinct tree sequences. Interoperation with Python simulators
+is then straightforward. For the proof-of-concept implementations discussed
```

Comment:
why not just this "the implementations described"?

Reply:
Yes, good idea. We can revisit this.